pkg/installmanager/installmanager.go

```diff
@@ -457,7 +458,31 @@ func runUninstaller(cd *hivev1.ClusterDeployment, infraID string, logger log.Fie
 			Logger:  logger,
 		}
 
-		return uninstaller.Run()
+		if err := uninstaller.Run(); err != nil {
+			return err
+		}
+
+		// If we're managing DNS for this cluster, lookup the DNSZone and cleanup
+		// any leftover A records that may have leaked due to
+		// https://jira.coreos.com/browse/CORS-1195.
+		if cd.Spec.ManageDNS {
+			dnsZone := &hivev1.DNSZone{}
+			dnsZoneNamespacedName := types.NamespacedName{Namespace: cd.Namespace, Name: controllerutils.DNSZoneName(cd.Name)}
+			err := dynClient.Get(context.TODO(), dnsZoneNamespacedName, dnsZone)
```

Comment:
Need to add permission [here](https://github.com/openshift/hive/blob/30a1011410d191c011ebba314c60c1b2ede677b7/pkg/controller/utils/sa.go#L27) in order to get DNSZone resources.

Reply:
Good catch, hadn't gotten far enough to hit that yet, thanks.